test/test_cmake/main.cpp

```diff
@@ -0,0 +1,192 @@
+// make sure we find all transitively included headers
+
+#include <boost/mpl/accumulate.hpp>
```

Comment:
It would be better to make a build directive to include each file separately and make sure it compiles, like the one in Boost.Uuid test jam, which I got from @lastique, I believe?

Reply:
I think, the point here is that it's better to avoid manual enumeration of the headers and generate it based on the filesystem. It's not exactly what we do to verify self-sufficiency of headers, but the implementation could be similar: generate a Boost.Preprocessor sequence of headers and expand it to includes in the .cpp. Or you could simply generate the entire .cpp file from CMake.

Reply:
@jeking3: So what is your opinion on this particular issue?